pkg/hooks/webhook.go

```diff
@@ -66,24 +76,61 @@ func NewWebhookExecutor(
 	if err != nil {
 		return nil, err
 	}
+	return NewWebhookExecutorWithCustomHttpClient(webhook, hookType, client)
+}
+
+func NewWebhookExecutorWithCustomHttpClient(
+	webhook *v1alpha1.Webhook,
+	hookType common.HookType,
+	httpClient HttpClientInterface,
+) (*WebhookExecutor, error) {
+	if webhook == nil {
```

Comment:
this is tested in `NewWebhookExecutor` - method which is invoking this one. Actually this method should not be exported (is used internally only) and should not repeat logic which is in callermethod

Reply:
made it unexportable, removed webhook verification and url calculation from there